src/cljs/ataru/hakija/application_hakukohde_component.cljs

```diff
@@ -4,37 +4,78 @@
     [ataru.application-common.application-field-common :refer [scroll-to-anchor]]
     [ataru.cljs-util :refer [get-translation]]))
 
-(defn index-of [s val from-index]
-  (clojure.string/index-of (clojure.string/lower-case s)
-                           (clojure.string/lower-case val)
-                           from-index))
-
 (defn- should-search? [search-term]
   (> (count search-term) 1))
 
-(defn match-text [text search-term]
-  (if-not (should-search? search-term)
+(defn text-matches
+  [text search-terms]
+  (let [text         (clojure.string/lower-case text)
+        search-terms (->> search-terms
+                          (filter should-search?)
+                          (map clojure.string/lower-case))]
+    (apply concat
+           (for [search-term search-terms]
+             (loop [res           []
+                    current-index 0]
+               (let [match-index (clojure.string/index-of text search-term current-index)
+                     match-end   (when match-index (+ match-index (count search-term)))]
+                 (if (nil? (and match-index match-end))
```

Comment:
Mä en erityisemmin diggaa `and`:n ja `or`:n ominaisuudesta, että ne palauttaa ei-boolean arvoja. Tässä lisäksi riittää katsoa `(nil? match-index)`. Tai vaihtoehtoisesti
```
(if-let [match-index (clojure.string/index-of text search-term current-index)]
  (recur (conj res [match-index (+ match-index (count search-term))]))
  res)
```

Reply:
Juu, muutan.